### ISSUE TITLE
can't run on my mac

### ISSUE THREAD

**Issue by proukakis** [songbowang125/SVision-pro#5]
hello thanks for testing this on mac, but it seems that my chip M1max may be an issue- or the OS 12.6

I get this error

Intel MKL FATAL ERROR: This system does not meet the minimum requirements for use of the Intel(R) Math Kernel Library.
The processor must support the Intel(R) Supplemental Streaming SIMD Extensions 3 (Intel(R) SSSE3) instructions.

Any suggestions?
thanks
christos

**Comment by proukakis** [comment]
so i did solve this pretty easily by 

conda install mkl 

but i get this now

Traceback (most recent call last):
  File "/Users/christos/miniconda3/envs/svision-pro-env/bin/SVision-pro", line 4, in <module>
    __import__('pkg_resources').run_script('SVision-pro==1.8', 'SVision-pro')
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 672, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1472, in run_script
    exec(code, namespace, namespace)
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg/EGG-INFO/scripts/SVision-pro", line 13, in <module>
    from src.output_vcf import output_vcf_final
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg/src/output_vcf.py", line 3, in <module>
    import pysam
ModuleNotFoundError: No module named 'pysam'


my command is 

![image](https://github.com/songbowang125/SVision-pro/assets/18488392/ad40ddd0-dea6-4035-b08f-423c4c9ee225)

**Comment by songbowang125** [comment]
Thanks for pointing out the potential failure when using a mac with M-series cores.  

_ModuleNotFoundError: No module named 'pysam'_  means the package 'pysam' is missed in your env. You can install it by 'pip install pysam'.

There are several packages required by SVision-pro (see environment.yml). In common, 'python setup.py install' would install all the packages, but if not, you might install manually.

**Comment by proukakis** [comment]
many thanks but it does seem to be there

pip install pysam
Requirement already satisfied: pysam in /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages (0.22.1)

and after running

python setup.py install

i get

running install
/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/setuptools/command/easy_install.py:147: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  EasyInstallDeprecationWarning,
running bdist_egg
running egg_info
writing SVision_pro.egg-info/PKG-INFO
writing dependency_links to SVision_pro.egg-info/dependency_links.txt
writing top-level names to SVision_pro.egg-info/top_level.txt
reading manifest file 'SVision_pro.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'SVision_pro.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.9-x86_64/egg
running install_lib
running build_py
creating build/bdist.macosx-10.9-x86_64/egg
creating build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/version.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/partition_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/color_plot.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/cigar_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/variant_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
creating build/bdist.macosx-10.9-x86_64/egg/src/hash_realign
copying build/lib/src/hash_realign/run_hash_realign.py -> build/bdist.macosx-10.9-x86_64/egg/src/hash_realign
copying build/lib/src/hash_realign/__init__.py -> build/bdist.macosx-10.9-x86_64/egg/src/hash_realign
copying build/lib/src/hash_realign/classes.py -> build/bdist.macosx-10.9-x86_64/egg/src/hash_realign
copying build/lib/src/hash_realign/hash_aligner.py -> build/bdist.macosx-10.9-x86_64/egg/src/hash_realign
copying build/lib/src/cluster_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
creating build/bdist.macosx-10.9-x86_64/egg/src/network
copying build/lib/src/network/op_dataset.py -> build/bdist.macosx-10.9-x86_64/egg/src/network
copying build/lib/src/network/model_unet.py -> build/bdist.macosx-10.9-x86_64/egg/src/network
copying build/lib/src/network/__init__.py -> build/bdist.macosx-10.9-x86_64/egg/src/network
creating build/bdist.macosx-10.9-x86_64/egg/src/network/utils
copying build/lib/src/network/utils/image_transforms.py -> build/bdist.macosx-10.9-x86_64/egg/src/network/utils
copying build/lib/src/network/utils/image_helpers.py -> build/bdist.macosx-10.9-x86_64/egg/src/network/utils
copying build/lib/src/network/utils/__init__.py -> build/bdist.macosx-10.9-x86_64/egg/src/network/utils
copying build/lib/src/network/model_liteunet.py -> build/bdist.macosx-10.9-x86_64/egg/src/network
copying build/lib/src/network/model_miniunet.py -> build/bdist.macosx-10.9-x86_64/egg/src/network
copying build/lib/src/__init__.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/cigar_map.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/collect_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/output_vcf.py -> build/bdist.macosx-10.9-x86_64/egg/src
creating build/bdist.macosx-10.9-x86_64/egg/src/pre_process
copying build/lib/src/pre_process/__init__.py -> build/bdist.macosx-10.9-x86_64/egg/src/pre_process
copying build/lib/src/inheritype_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
copying build/lib/src/align_op.py -> build/bdist.macosx-10.9-x86_64/egg/src
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/version.py to version.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/partition_op.py to partition_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/color_plot.py to color_plot.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/cigar_op.py to cigar_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/variant_op.py to variant_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/hash_realign/run_hash_realign.py to run_hash_realign.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/hash_realign/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/hash_realign/classes.py to classes.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/hash_realign/hash_aligner.py to hash_aligner.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/cluster_op.py to cluster_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/op_dataset.py to op_dataset.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/model_unet.py to model_unet.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/utils/image_transforms.py to image_transforms.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/utils/image_helpers.py to image_helpers.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/utils/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/model_liteunet.py to model_liteunet.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/network/model_miniunet.py to model_miniunet.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/cigar_map.py to cigar_map.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/collect_op.py to collect_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/output_vcf.py to output_vcf.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/pre_process/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/inheritype_op.py to inheritype_op.cpython-37.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/src/align_op.py to align_op.cpython-37.pyc
installing package data to build/bdist.macosx-10.9-x86_64/egg
running install_data
copying LICENSE -> build/bdist.macosx-10.9-x86_64/egg/
creating build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
installing scripts to build/bdist.macosx-10.9-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.macosx-10.9-x86_64/egg/EGG-INFO/scripts
copying build/scripts-3.7/SVision-pro -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.macosx-10.9-x86_64/egg/EGG-INFO/scripts/SVision-pro to 755
copying SVision_pro.egg-info/PKG-INFO -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
copying SVision_pro.egg-info/SOURCES.txt -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
copying SVision_pro.egg-info/dependency_links.txt -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
copying SVision_pro.egg-info/not-zip-safe -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
copying SVision_pro.egg-info/top_level.txt -> build/bdist.macosx-10.9-x86_64/egg/EGG-INFO
creating 'dist/SVision_pro-1.8-py3.7.egg' and adding 'build/bdist.macosx-10.9-x86_64/egg' to it
removing 'build/bdist.macosx-10.9-x86_64/egg' (and everything under it)
Processing SVision_pro-1.8-py3.7.egg
removing '/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg' (and everything under it)
creating /Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg
Extracting SVision_pro-1.8-py3.7.egg to /Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages
SVision-pro 1.8 is already the active version in easy-install.pth
Installing SVision-pro script to /Users/christos/miniconda3/envs/svision-pro-env/bin

Installed /Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg
Processing dependencies for SVision-pro==1.8
Finished processing dependencies for SVision-pro==1.8

but is still fails

Traceback (most recent call last):
  File "/Users/christos/miniconda3/envs/svision-pro-env/bin/SVision-pro", line 4, in <module>
    __import__('pkg_resources').run_script('SVision-pro==1.8', 'SVision-pro')
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 672, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1472, in run_script
    exec(code, namespace, namespace)
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg/EGG-INFO/scripts/SVision-pro", line 13, in <module>
    from src.output_vcf import output_vcf_final
  File "/Users/christos/miniconda3/envs/svision-pro-env/lib/python3.7/site-packages/SVision_pro-1.8-py3.7.egg/src/output_vcf.py", line 3, in <module>
    import pysam
ModuleNotFoundError: No module named 'pysam'

It is possible i am doing something really stupid, but i can't work out what

**Comment by songbowang125** [comment]
It is indeed a bit strange. I will attempt the installation process later using an M1 Mac and will send you the results.

**Comment by proukakis** [comment]
ok, many thanks

**Comment by proukakis** [comment]
So somehow some of the requirements were not in the env

I did conda install for pysam, scipy, opencv

then it started running but had another error:

OMP: Error #15: Initializing libomp.dylib, but found libiomp5.dylib already initialized.                                                                                                                          OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://openmp.llvm.org/
Abort trap: 6                    

working on  this now

**Comment by proukakis** [comment]
so, it looks like it's working, wasn't simple, hopefully will help others

i had to 

conda remove mkl

then reinstall many things as was getting successive error messages

Conda update --all

Conda install --numpy

Conda install --opencv

conda install pytorch

Conda install pillow

now time to have fun!

**Comment by songbowang125** [comment]
Happy to know that. 

While we have just tried the install process on a M-chip Mac, and there was no any failure.

We just followed the install process described on the GitHub page, which is 

```
conda env create -f ./environment.yml 
conda activate svision-pro-env
python setup.py install
```

Perhaps, the basic running env is different between mine and yours. For example, the conda has both Intel version and M version, but I'm not entirely sure.

We really appreciate your attempt. It will definitely be helpful for others who might encounter the same issue.
